src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,72 @@
+package parser;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import shop.Cart;
+
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class JsonParserTest {
+
+    JsonParser testJsonParser;
+
+    @BeforeEach
+    void setUp() {
+       testJsonParser = new JsonParser();
+    }
+
+    @AfterEach
+    void tearDown() {
+        testJsonParser = null;
+    }
+
+//test checks that method could use various cart names with symbols
+    @ParameterizedTest
+    @ValueSource(strings = {"testCart", "!@#cart", "$%^cart", "@^&cart", "123&()cart"})
+    void writeToFile(String cartName) {
+        Cart testCart = new Cart(cartName);
+        testJsonParser.writeToFile(testCart);
+        File testFile = new File("src\\main\\resources\\"+cartName+".json");
+        assertEquals(cartName, testJsonParser.readFromFile(testFile).getCartName());
+        testFile.delete();
+    }
+
+//test checks that corresponding input parameters could cause NoSuchFileException and JsonSyntaxException
+    @ParameterizedTest
+    @ValueSource(strings = {"src\\main\\resources\\fake", "src\\main\\resources\\f1.json"})
+    void readFromFile(String fileName) {
+
```

Comment:
Do not add blank line in the beginning of the method. I try to follow standard java code convention during the training (it can be a little bit differ among projects)